src/auth/auth.service.ts

```diff
@@ -102,6 +114,37 @@ export class AuthService {
     return user;
   }
 
+  async generateAccountActivationToken(userId: number): Promise<string> {
+    const token = await this.generateBearerToken(
+      userId,
+      process.env.JWT_ACCOUNT_ACTIVATION_SECRET,
+      +process.env.ACCOUNT_ACTIVATION_TIME_IN_SECONDS,
+    );
+    await this.userRepository.saveAccountActivationToken(userId, token);
+
+    return token;
+  }
+
+  async verifyAccountActivationToken(
+    jwtToken: string,
+  ): Promise<{ id: number }> {
+    try {
+      return this.jwtService.verifyAsync(jwtToken, {
+        secret: process.env.JWT_ACCOUNT_ACTIVATION_SECRET,
+      });
+    } catch {
+      throw new ForbiddenException('Token expired');
+    }
+  }
+
+  async activateAccount(userId: number): Promise<UserPayloadRequest> {
+    const userData = await this.userRepository.getPendingUserById(userId);
+
+    await this.userRepository.removePendingUserById(userId);
+
+    return this.userRepository.createUser(userData);
```

Comment:
hmm, not that this is wrong, but I'd rather first create resources, then remove old ones.

Also a note (you don't have to do this) - using a raw SQL you should probably be able to copy data from pendingUser to user without loading them into the backend